# 0. Enter to drive

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
%cd /content/drive/MyDrive/Github/Hackathon-BBVA-2022
!ls -a

/content/drive/MyDrive/Github/Hackathon-BBVA-2022
CHANGELOG.rst  .git	   House_prediction.ipynb  Models     requirements.txt
Dataset        .gitignore  LICENSE		   README.md


## 0.1. Connect to Github

In [ ]:
!git checkout enzo

In [ ]:
!git status

In [ ]:
!git add .gitignore
!git add CHANGELOG.rst
!git add Dataset
!git add House_prediction.ipynb
!git add README.md
!git add requirements.txt

In [ ]:
!git config --global user.email "enzo.cisneros@pucp.edu.pe"

In [171]:
!git commit -m "0.0.2"

[enzo 1f839bd] 0.0.2
 1 file changed, 1 insertion(+), 1 deletion(-)
 rewrite House_prediction.ipynb (96%)


In [172]:
!git status

On branch enzo
Your branch is ahead of 'origin/enzo' by 1 commit.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean


In [173]:
!git push --set-upstream origin enzo

Counting objects: 3, done.
Delta compression using up to 2 threads.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 9.31 KiB | 1.55 MiB/s, done.
Total 3 (delta 1), reused 0 (delta 0)
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To https://github.com/ECisneros20/Hackathon-BBVA-2022
   135b384..1f839bd  enzo -> enzo
Branch 'enzo' set up to track remote branch 'enzo' from 'origin'.


In [174]:
!git pull

remote: Enumerating objects: 10, done.
remote: Counting objects: 100% (2/2), done.
remote: Total 10 (delta 2), reused 2 (delta 2), pack-reused 8
Unpacking objects: 100% (10/10), done.
From https://github.com/ECisneros20/Hackathon-BBVA-2022
 * [new branch]      Julio      -> origin/Julio
   e6b2125..7a17c99  luffy      -> origin/luffy
   f045399..2af8553  testing    -> origin/testing
Already up to date.


# 1. Data preprocessing

## 1.1. Load data

In [ ]:
#!gzip -dk Dataset/pe_properties.csv.gz

In [3]:
import pandas as pd
pd.set_option('mode.chained_assignment', None)

peru = pd.read_csv('Dataset/pe_properties.csv')

In [4]:
print("Original dataset shape: ", peru.shape, "\n")
peru.head()

Original dataset shape:  (100624, 25) 



id    ad_type  start_date    end_date  created_on  \
0  1qOI7X9aqAcO7PzjX1B29Q==  Propiedad  2021-06-03  2021-08-27  2021-06-03   
1  J350rcSI7Fx+92Wb1ev7rw==  Propiedad  2021-06-03  2021-07-07  2021-06-03   
2  P+3USxoY2w5HQwg1N8JCtg==  Propiedad  2021-06-03  9999-12-31  2021-06-03   
3  I9llUYb61GVgLNDUJ6CT5A==  Propiedad  2021-06-03  2021-06-04  2021-06-03   
4  d7Sw3mL1Vq0dmzoXNpJUNQ==  Propiedad  2021-06-03  2021-06-04  2021-06-03   

         lat       lon    l1           l2        l3  ... bathrooms  \
0        NaN       NaN  Perú         Lima      Lima  ...       5.0   
1        NaN       NaN  Perú     Arequipa  Arequipa  ...       6.0   
2  -8.095431 -79.05310  Perú  La Libertad  Trujillo  ...       4.0   
3 -13.840915 -76.25083  Perú          Ica       Ica  ...       NaN   
4 -12.799091 -76.56912  Perú         Lima      Lima  ...       NaN   

  surface_total surface_covered      price  currency  price_period  \
0           NaN             NaN   450000.0       USD           NaN   
1           NaN             NaN  1050000.0       USD           NaN   
2           NaN             NaN   179000.0       USD           NaN   
3        1120.0             NaN    52000.0       USD       Mensual   
4         500.0             NaN    70000.0       USD       Mensual   

                                               title  \
0                 CASA DE PLAYA VENDO EN LAS LAGUNAS   
1         HERMOSA CASA DE ENSUEÑOS EN CERRO COLORADO   
2  [ VENTA ]: ESTRENA TU CASA EN URB. SOL Y LUZ C...   
3                                         Lote venta   
4                                         Lote venta   

                                         description  property_type  \
0  <b>CASA DE PLAYA VENDO EN LAS LAGUNAS</b><br><...           Casa   
1  <b>HERMOSA CASA DE ENSUEÑOS EN CERRO COLORADO<...           Casa   
2  <b>[ VENTA ]: ESTRENA TU CASA EN URB. SOL Y LU...           Casa   
3  Hábitat VIp vende exclusivo terreno 1120 m2  e...           Lote   
4  Hábitat Vip Vende lote 500 m2 \r\nUbicado a ta...           Lote   

  operation_type  
0          Venta  
1          Venta  
2          Venta  
3          Venta  
4          Venta  

[5 rows x 25 columns]

In [5]:
peru.columns

Index(['id', 'ad_type', 'start_date', 'end_date', 'created_on', 'lat', 'lon',
       'l1', 'l2', 'l3', 'l4', 'l5', 'l6', 'rooms', 'bedrooms', 'bathrooms',
       'surface_total', 'surface_covered', 'price', 'currency', 'price_period',
       'title', 'description', 'property_type', 'operation_type'],
      dtype='object')

In [6]:
print("Dataset shape: ", peru.shape, "\n")

for i in peru.columns:
  print(i, "count ->", peru[i].nunique())
  print(i, "nan ->", peru[i].isnull().sum())
  print(i, "nan ->", round(peru[i].isnull().sum() / peru.shape[0] * 100, 2), "%")
  if peru[i].nunique() < 10:
    print(i, "values ->", peru[i].unique())
  else:
    print(i, "values ->", peru[i].unique()[0:4])
  print("\n")

Dataset shape:  (100624, 25) 

id count -> 100624
id nan -> 0
id nan -> 0.0 %
id values -> ['1qOI7X9aqAcO7PzjX1B29Q==' 'J350rcSI7Fx+92Wb1ev7rw=='
 'P+3USxoY2w5HQwg1N8JCtg==' 'I9llUYb61GVgLNDUJ6CT5A==']


ad_type count -> 1
ad_type nan -> 0
ad_type nan -> 0.0 %
ad_type values -> ['Propiedad']


start_date count -> 388
start_date nan -> 0
start_date nan -> 0.0 %
start_date values -> ['2021-06-03' '2021-06-17' '2020-08-31' '2021-07-24']


end_date count -> 450
end_date nan -> 0
end_date nan -> 0.0 %
end_date values -> ['2021-08-27' '2021-07-07' '9999-12-31' '2021-06-04']


created_on count -> 388
created_on nan -> 0
created_on nan -> 0.0 %
created_on values -> ['2021-06-03' '2021-06-17' '2020-08-31' '2021-07-24']


lat count -> 28526
lat nan -> 14006
lat nan -> 13.92 %
lat values -> [        nan  -8.0954308 -13.8409149 -12.799091 ]


lon count -> 28247
lon nan -> 14006
lon nan -> 13.92 %
lon values -> [         nan -79.05310039 -76.2508304  -76.56912   ]


l1 count -> 1
l1 nan -> 0
l1 nan

### Drop columns:
- **id**: same amount as index
- **ad_type**: only one value
- **start_date**: not relevant
- **end_date**: not relevant
- **created_on**: not relevant
- **l1**: only one value
- **l6**: 99.17% nan - too many nan values
- **rooms**: 97.82% nan - too many nan values
- **surface_total**: 82.85% nan - too many nan values
- **surface_covered**: 81.18% nan - too many nan values
- **price_period**: only one value
- **title**: not relevant
- **description**: not relevant
- **operation_type** = [Alquiler, Alquiler temporal]






### Keep columns:
- **lat**: 13.92% nan
- **lon**: 13.92% nan
- **l2**: 0.00% nan
- **l3**: 3.44% nan
- **l4**: 22.03% nan
- **l5**: 54.34% nan - too many nan values
- **bedrooms**: 58.05% nan - too many nan values
- **bathrooms**: 23.75% nan
- **price**: 0.65% nan
- **currency**: 0.72% nan
- **property_type**: 0.00% nan
- **operation_type** = Venta



## 1.2. Drop not necessary columns

In [7]:
peru.drop(["id", "ad_type", "start_date", "end_date", "created_on", "l1", "l6", "rooms", "surface_total", "surface_covered", "price_period", "title", "description"], axis = 1, inplace = True)
peru = peru.loc[peru["operation_type"] == "Venta"]
peru.drop("operation_type", axis = 1, inplace = True)
print("New dataset shape: ", peru.shape, "\n")
peru.head()

New dataset shape:  (68487, 11) 



lat       lon           l2        l3         l4   l5  bedrooms  \
0        NaN       NaN         Lima      Lima        NaN  NaN       4.0   
1        NaN       NaN     Arequipa  Arequipa  Yanahuara  NaN       4.0   
2  -8.095431 -79.05310  La Libertad  Trujillo        NaN  NaN       5.0   
3 -13.840915 -76.25083          Ica       Ica        NaN  NaN       NaN   
4 -12.799091 -76.56912         Lima      Lima        NaN  NaN       NaN   

   bathrooms      price currency property_type  
0        5.0   450000.0      USD          Casa  
1        6.0  1050000.0      USD          Casa  
2        4.0   179000.0      USD          Casa  
3        NaN    52000.0      USD          Lote  
4        NaN    70000.0      USD          Lote

## 1.3. Change prices to only USD currency

In [8]:
peru["currency"].value_counts()

USD    64254
PEN     3682
Name: currency, dtype: int64

In [9]:
change_PEN_to_USD = 1 / 4

peru["price"][peru["currency"] == "PEN"] = peru["price"] * change_PEN_to_USD
peru.drop("currency", axis = 1, inplace = True)
print("New dataset shape: ", peru.shape, "\n")
peru.head()

New dataset shape:  (68487, 10) 



lat       lon           l2        l3         l4   l5  bedrooms  \
0        NaN       NaN         Lima      Lima        NaN  NaN       4.0   
1        NaN       NaN     Arequipa  Arequipa  Yanahuara  NaN       4.0   
2  -8.095431 -79.05310  La Libertad  Trujillo        NaN  NaN       5.0   
3 -13.840915 -76.25083          Ica       Ica        NaN  NaN       NaN   
4 -12.799091 -76.56912         Lima      Lima        NaN  NaN       NaN   

   bathrooms      price property_type  
0        5.0   450000.0          Casa  
1        6.0  1050000.0          Casa  
2        4.0   179000.0          Casa  
3        NaN    52000.0          Lote  
4        NaN    70000.0          Lote

## 1.4. (TO DO) Check again nan values

In [10]:
print("Dataset shape: ", peru.shape, "\n")

for i in peru.columns:
  print(i, "count ->", peru[i].nunique())
  print(i, "nan ->", peru[i].isnull().sum())
  print(i, "nan ->", round(peru[i].isnull().sum() / peru.shape[0] * 100, 2), "%")
  if peru[i].nunique() < 10:
    print(i, "values ->", peru[i].unique())
  else:
    print(i, "values ->", peru[i].unique()[0:4])
  print("\n")

Dataset shape:  (68487, 10) 

lat count -> 19614
lat nan -> 9422
lat nan -> 13.76 %
lat values -> [        nan  -8.0954308 -13.8409149 -12.799091 ]


lon count -> 19415
lon nan -> 9422
lon nan -> 13.76 %
lon values -> [         nan -79.05310039 -76.2508304  -76.56912   ]


l2 count -> 25
l2 nan -> 0
l2 nan -> 0.0 %
l2 values -> ['Lima' 'Arequipa' 'La Libertad' 'Ica']


l3 count -> 91
l3 nan -> 2670
l3 nan -> 3.9 %
l3 values -> ['Lima' 'Arequipa' 'Trujillo' 'Ica']


l4 count -> 220
l4 nan -> 16885
l4 nan -> 24.65 %
l4 values -> [nan 'Yanahuara' 'Piura' 'Lima Centro']


l5 count -> 60
l5 nan -> 38213
l5 nan -> 55.8 %
l5 values -> [nan 'Lince' 'Rímac' 'La Molina']


bedrooms count -> 24
bedrooms nan -> 40094
bedrooms nan -> 58.54 %
bedrooms values -> [ 4.  5. nan  3.]


bathrooms count -> 19
bathrooms nan -> 20467
bathrooms nan -> 29.88 %
bathrooms values -> [ 5.  6.  4. nan]


price count -> 4879
price nan -> 510
price nan -> 0.74 %
price values -> [ 450000. 1050000.  179000.   52000.]



## 1.5. (TO DO) Impute data